tcms/utils/permissions.py

```diff
@@ -3,30 +3,46 @@
 from django.contrib.auth.models import Group, Permission
 
 
-def assign_default_group_permissions():
+def assign_default_group_permissions(output=None, refresh_all=False):
     """
         Adds the default permissions for Administrator and Tester
         groups!
     """
     admin = Group.objects.get(name='Administrator')
-    if admin.permissions.count() == 0:
+    if output:
+        output.write('\nAdding permissions to Administrator group')
+    if admin.permissions.count() == 0 or refresh_all:
         all_perms = Permission.objects.all()
-        admin.permissions.add(*all_perms)
+        admin_perms = admin.permissions.all()
+        perms_to_add = all_perms.exclude(pk__in=admin_perms)
+        admin.permissions.add(*perms_to_add)
+        admin.save()
```

Comment:
`admin.permissions` is an object manager, e.g. works directly with SQL under the hood. 
`admin.save()` is a method on the object which doesn't have any relation to the many-to-many fields. It's not necessary.